### PR TITLE
ENH Add environment variable to control 'exports' cli argument

### PR DESCRIPTION
**Pull request by hoodmane**
If "--exports" is not passed as a CLI argument, check for an environment variable called PYODIDE_EXPORTS. If this is also missing, default to "requested".

cc @henryiii 

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

File: pyodide-build/pyodide_build/io.py
Comment (hoodmane) [inline pyodide-build/pyodide_build/io.py:90]: This is awkward but we need it until typer merges something like this: https://github.com/tiangolo/typer/pull/429

**Comment by henryiii** [comment]
Not at computer, but Typer (if that’s what you are using) has integrated support for envvars: https://typer.tiangolo.com/tutorial/arguments/envvar/ (most CLI parsers do).

**Comment by hoodmane** [comment]
Ah great thanks for pointing that out @henryiii.

**Comment by rth** [comment]
Thanks! Could you also document it under https://pyodide.org/en/stable/development/building-from-sources.html#environment-variables ? Maybe it would make sense to have something more specific? Like `PYODIDE_BUILD_EXPORTS` to indicate that it's related to the `pyodide build` command. There are a lot of things that could be exports in different contexts. Otherwise LGTM.

**Comment by hoodmane** [comment]
@rth @ryanking13 could one of you give this a quick review? I want to backport it to 0.23.4 and it's the last thing I want to add to that release.

File: pyodide-build/pyodide_build/pypabuild.py
Comment (ryanking13) [inline pyodide-build/pyodide_build/pypabuild.py:189]: Maybe we can define a type for `_BuildSpecExports | list[str]`.  It is used everywhere in this PR.

File: pyodide-build/pyodide_build/cli/build.py
Comment (ryanking13) [inline pyodide-build/pyodide_build/cli/build.py:31]: Maybe:

```py
from typing import get_args
possible_exports = get_args(_BuildSpecExports)
if exports in possible_exports:
    return exports
```

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane! and sorry for the late review. I have some minor comments otherwise LGTM. Please also update the changelog.

File: pyodide-build/pyodide_build/cli/build.py
Comment (hoodmane) [reply pyodide-build/pyodide_build/cli/build.py:31]: I think this pattern doesn't convince mypy? You could say `return cast(_BuildSpecExports, exports)`.

Comment (hoodmane) [reply pyodide-build/pyodide_build/cli/build.py]: Thanks for the input @henryiii and @ryanking13!

**Comment by rth** [comment]
Nice! LGTM as well and CI is green.

**Pull request merged**
